lib/rtm-client.js

```diff
@@ -89,11 +101,40 @@ class RtmClient extends EventEmitter {
   }
 
   subscribe({channel, options}) {
-    return promised(cb => this.mqttClient.subscribe(channel, options, cb));
+    if (this.isSubscribedTo(channel)) {
+      return Promise.resolve();
+    }
+
+    return Promise.all([
```

Comment:
I think the internal `subscribedChannels` cache should be updated **after** subscribe finished:

```js
return promised(cb => this.mqttClient.subscribe(channel, options, cb))
  .then(() => {
    this.subscribedChannels.push(channel);
  })
```